### PR TITLE
feat(Storybook): extend storybook for developer focused demo - #315

### PR DESCRIPTION
**Pull request by 98lenvi**
# Issue #315 #376 
Complete support added for `Cicero-UI` components in `Storybook`. 

### Changes
- Extended `webpackconfig` of `storybook`'s default configuartion
  - Added sass loader
  - Excluding `jsdom`
  - Added config to ignore core Node APIs (`fs`,`net`,`tls`) during build.
- Moved storybook and related dependencies to dev-dependencies. 
- Added `netlify.toml` to netlify to storybook build

Previously added, now removed-
  - Added `CleanWebpackPlugin()`
  - Added `HtmlWebpackPlugin`
  - Added `MiniCssExtractPlugin`

### Flags
Need to add the following (DONE)
- [x]  ContractEditor
- [x]  TemplateLibrary
- [x]  ErrorLogger
- [x]  Navigation
- [x]  Demo

### Related Issues
- Issue #315 
- Issue #356 
- Issue #376 
- Pull Request #353 
- Pull Request #335 
- Pull Request #358

### CODE REVIEW

**Comment by 98lenvi** [comment]
@irmerk, I hope it's complete now. Can you please go through it and let me know if anything needs to be added / modified

File: .storybook/main.js [hunk 1]
```diff
@@ -1,10 +1,36 @@
...
+      test: /\.scss$/,
+      use: ["style-loader", "css-loader", "sass-loader"],
+      include: path.resolve(__dirname, "../")
+    });
+    config.plugins.push(new MiniCssExtractPlugin({ filename: "[name].css" }));
+    config.plugins.push(
+      new HtmlWebpackPlugin({
+        template: "demo/src/index.html",
+        mountId: "demo",
+        title: "YES"
```
Comment (DianaLease) [inline .storybook/main.js:19]: Can we make the title something like "Cicero UI Components Demo"?

File: demo/src/index.js [hunk 2]
```diff
@@ -203,3 +203,6 @@ function Demo() {
 }
 
 render(<Demo/>, document.querySelector('#root'));
+
+//For usage with Storybook (temporary)
```
Comment (DianaLease) [inline demo/src/index.js:207]: What does this comment mean? Is it temporary because we will make a new "Demo" component for storybook instead of using the existing one?

Comment (98lenvi) [reply demo/src/index.js:207]: I had added lines 207 & 208 in this [commit](https://github.com/accordproject/cicero-ui/pull/382/commits/f08ec5c0dda94393d63bd50bb319797ca0dedfdf). It was added to export the Demo component to be used with storybook for testing. 

I have now reverted the changes (i.e. I have now removed line 207 & 208) now as Demo is no longer needed in Storybook.

File: .storybook/main.js (hunk 1 above)
Comment (98lenvi) [reply .storybook/main.js:19]: Sure, Will do it right away!

**Comment by 98lenvi** [comment]
> Added a few comments. Also, is there a way to see the netlify build of storybook for this PR?

@DianaLease  I will be adding a `netlify.toml` file for that in a commit.

**Comment by irmerk** [comment]
This is super exciting @98lenvi, great work. Looking into the code now.

**Comment by irmerk** [comment]
In the Notes tab for most of these, we'll want to make sure to format some things to make it consistent with the `.md` format here in the GitHub repo. Example is the main Demo which has the `README.md` from the main repo here, and the table for Ecosystem > Core libraries is a bit off.

**Comment by dselman** [comment]
> In the Notes tab for most of these, we'll want to make sure to format some things to make it consistent with the `.md` format here in the GitHub repo. Example is the main Demo which has the `README.md` from the main repo here, and the table for Ecosystem > Core libraries is a bit off.

Let's try to avoid duplication as much as possible to keep things maintainable. E.g. link to the existing README.md.

File: netlify.toml [hunk 3]
```diff
@@ -0,0 +1,43 @@
+# Settings in the [build] context are global and are applied to all contexts
```
Comment (jeromesimeon) [inline netlify.toml:1]: Do we need a netlify configuration? We don't do that on any of our projects -- should it be instead done within netlify itself.

Comment (98lenvi) [reply netlify.toml:1]: Yes, it can be done in Netlify instead.

**Comment by 98lenvi** [comment]
> In the Notes tab for most of these, we'll want to make sure to format some things to make it consistent with the `.md` format here in the GitHub repo. Example is the main Demo which has the `README.md` from the main repo here, and the table for Ecosystem > Core libraries is a bit off.

I will look into that, can't understand why the formatting looks off. @irmerk 


> Let's try to avoid duplication as much as possible to keep things maintainable. E.g. link to the existing README.md.

Done in the latest commit. @dselman

**Comment by irmerk** [comment]
I'm thinking we get this in and start iterating on it to stop blocking this PR. Just need to make sure local development of `ContractEditor` still works as we expect.

**Comment by 98lenvi** [comment]
@irmerk Sure! I just need to work on one more commit to add knobs to ContractEditor story ( I missed the lockText and readOnly props)

**Comment by 98lenvi** [comment]
@irmerk, I made the changes, i.e. add readOnly knob. For some reason, lockText doesn't work via a knob. Even if I declare it as a state variable and change its value, there is no effect on the ContractEditor.

**Comment by DianaLease** [comment]
> @irmerk, I made the changes, i.e. add readOnly knob. For some reason, lockText doesn't work via a knob. Even if I declare it as a state variable and change its value, there is no effect on the ContractEditor.

This sounds similar to an issue we had with the old version of slate where prop changes wouldn't re-render the editor. It was working for me with the new slate when we had a toggle switch though that updated the value of lockText. Let me try again locally with a toggle and reconfirm.

**Comment by irmerk** [comment]
Confirming this works for local development (`npm run storybook` instead of `npm run start`)
